src/test/fuzz/tx_pool.cpp

```diff
@@ -239,33 +251,52 @@ FUZZ_TARGET_INIT(tx_pool_standard, initialize_tx_pool)
+                // A package must have at least 2 txs, so size-1 packages can only be submitted standalone
+                package.size() == 1 ||
+                fuzzed_data_provider.ConsumeBool()};
+            if (submit_standalone) {
+                const auto result_single{WITH_LOCK(::cs_main, return AcceptToMemoryPool(chainstate, tx, GetTime(), bypass_limits, /*test_accept=*/false))};
+                const bool accepted_single{result_single.m_result_type == MempoolAcceptResult::ResultType::VALID};
+                accepted &= accepted_single;
+            }
         }
 
-        const auto res = WITH_LOCK(::cs_main, return AcceptToMemoryPool(chainstate, tx, GetTime(), bypass_limits, /*test_accept=*/false));
-        const bool accepted = res.m_result_type == MempoolAcceptResult::ResultType::VALID;
+        if (package.size() >= 2) {
+            // Submit the remainder of the package.
+            // For now, every package has only one tx , so this submits the whole
+            // package again.
+            const auto result_package{WITH_LOCK(::cs_main, return ProcessNewPackage(chainstate, tx_pool, package, /*test_accept=*/false))};
+            accepted = result_package.m_state.IsValid();
+            Assert(accepted != result_package.m_state.IsInvalid());
+            // If something went wrong due to a package-specific policy, it might not return a
+            // validation result for the transaction.
+            if (result_package.m_state.GetResult() != PackageValidationResult::PCKG_POLICY) {
+                for (const auto& tx_res : result_package.m_tx_results) {
+                    Assert(tx_res.second.m_result_type == MempoolAcceptResult::ResultType::VALID ||
+                           tx_res.second.m_result_type == MempoolAcceptResult::ResultType::INVALID);
+                }
+            }
+            if (accepted) {
+                Assert(result_package.m_tx_results.size() == package.size());
+                for (const auto& tx_res : result_package.m_tx_results) {
+                    Assert(tx_res.second.m_result_type == MempoolAcceptResult::ResultType::VALID);
+                }
+            }
+        }
         SyncWithValidationInterfaceQueue();
         UnregisterSharedValidationInterface(txr);
-
         Assert(accepted != added.empty());
-        Assert(accepted == res.m_state.IsValid());
-        Assert(accepted != res.m_state.IsInvalid());
         if (accepted) {
-            Assert(added.size() == 1); // For now, no package acceptance
-            Assert(tx == *added.begin());
+            Assert(Package(added.begin(), added.end()) == package);
         } else {
-            // Do not consider rejected transaction removed
-            removed.erase(tx);
+            // Do not consider rejected transactions removed
+            for (const auto& tx : package) {
+                removed.erase(tx);
+            }
         }
 
         // Helper to insert spent and created outpoints of a tx into collections
```

Comment:
Another bug assert violation happens in `insert_tx`.

```log
test/fuzz/tx_pool.cpp:308 operator(): Assertion `set.get().emplace(tx.GetHash(), i).second' failed.
==3192931== ERROR: libFuzzer: deadly signal
    #0 0x55c5a9cbf680 in __sanitizer_print_stack_trace (/home/ubuntu/bitcoin/src/test/fuzz/fuzz+0xfd7680)
    #1 0x55c5a9c6b988 in fuzzer::PrintStackTrace() (/home/ubuntu/bitcoin/src/test/fuzz/fuzz+0xf83988)
    #2 0x55c5a9c50ad3 in fuzzer::Fuzzer::CrashCallback() (/home/ubuntu/bitcoin/src/test/fuzz/fuzz+0xf68ad3)
    #3 0x7f666ad8a41f  (/lib/x86_64-linux-gnu/libpthread.so.0+0x1441f)
    #4 0x7f666a9f000a in __libc_signal_restore_set /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/internal-signals.h:86:3
    #5 0x7f666a9f000a in raise /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/raise.c:48:3
    #6 0x7f666a9cf858 in abort /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:79:7
    #7 0x55c5abfff51d in assertion_fail(char const*, int, char const*, char const*) src/util/check.cpp:13:5
    #8 0x55c5aa21ec02 in bool&& inline_assertion_check<true, bool>(bool&&, char const*, int, char const*, char const*) src/./util/check.h:67:13
    #9 0x55c5aa4d3759 in auto (anonymous namespace)::tx_pool_standard_fuzz_target(Span<unsigned char const>)::$_1::operator()<CTransaction>(std::vector<std::reference_wrapper<std::set<COutPoint, std::less<COutPoint>, std::allocator<COutPoint> > >, std::allocator<std::reference_wrapper<std::set<COutPoint, std::less<COutPoint>, std::allocator<COutPoint> > > > >, std::vector<std::reference_wrapper<std::set<COutPoint, std::less<COutPoint>, std::allocator<COutPoint> > >, std::allocator<std::reference_wrapper<std::set<COutPoint, std::less<COutPoint>, std::allocator<COutPoint> > > > >, CTransaction const&) const src/test/fuzz/tx_pool.cpp:308:21
    #10 0x55c5aa4cdf0a in (anonymous namespace)::tx_pool_standard_fuzz_target(Span<unsigned char const>) src/test/fuzz/tx_pool.cpp:327:17
```

I am not sure about what the loops are doing. Maybe `outpoint_rbf` should not be modified when constructing transactioins.

Reply:
This is solved in https://github.com/bitcoin/bitcoin/commit/6ca7fcf2247ed7067ebaf3ee8dc325d49142d666 by use a new copy of `outpoint_rbf` in the loop.